src/Frontend/Modules/Profiles/Engine/Model.php

```diff
@@ -290,57 +268,25 @@ public static function getUrl(string $displayName, int $excludedId = null): stri
         // urlise
         $url = CommonUri::getUrl($displayName);
 
-        // get database
-        $database = FrontendModel::getContainer()->get('database');
-
-        // new item
-        if ($excludedId === null) {
-            // get number of profiles with this URL
-            $number = (int) $database->getVar(
-                'SELECT 1
-                 FROM profiles AS p
-                 WHERE p.url = ?
-                 LIMIT 1',
-                (string) $url
-            );
-
-            // already exists
-            if ($number !== 0) {
-                // add number
-                $url = FrontendModel::addNumber($url);
-
-                // try again
-                return self::getUrl($url);
-            }
-
-            return $url;
-        }
-
-        // current profile should be excluded
-        // get number of profiles with this URL
-        $number = (int) $database->getVar(
-            'SELECT 1
-             FROM profiles AS p
-             WHERE p.url = ? AND p.id != ?
-             LIMIT 1',
-            [$url, $excludedId]
+        return FrontendModel::get('profile.repository.profile')->getUrl(
+            $url,
+            $excludedId
         );
-
-        // already exists
-        if ($number !== 0) {
-            // add number
-            $url = FrontendModel::addNumber($url);
-
-            // try again
-            return self::getUrl($url, $excludedId);
-        }
-
-        return $url;
     }
 
     public static function insert(array $profile): int
     {
-        return (int) FrontendModel::getContainer()->get('database')->insert('profiles', $profile);
+        $profile = new Profile(
```

Comment:
I would use a different variable name, cause you are turning the array into an Profile class. While possible I would still try to avoid it

Reply:
Fixed 7e48a7326